tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -55,112 +55,132 @@ steps:
       echo Tag: $tag
       cp .npmrc ~/.npmrc
       maximumRetryIfNetworkError=3
+
+      # Note: the package names here are "processed" the way 'npm pack' does (if the package is scoped, remove
+      # the leading @ and replace / with -).
+      # When getting them from a txt file that's the way we wrote them to it.
+      # When getting them from the output of 'ls' that's also how they are because the files are the 'npm pack'ed
+      # packages but we need to strip the version number and the extension for consistency, so later we can assume
+      # that these are just the package names, as we need them like that for some things.
       if [[ -f $(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt ]]; then
-        packages=$(cat $(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt)
-        sorted=true
+        processedPackageNames=$(cat $(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt)
       else
-        packages=*.tgz
-        sorted=false
+        # The sed expression removes from the first dash followed by a number, to the file extension.
+        # This assumes we don't have packages whose names have a dash followed by a number.
+        # See comment above for more details.
+        processedPackageNames=$(ls *.tgz | sed 's/-[0-9].*\.tgz//g')
       fi
-      for packageName in $packages
+      for processedPackageName in $processedPackageNames
       do
-        if [[ $sorted == true ]]; then
```

Comment:
The `$sorted` variable unnecessary since now we have to look for the file based on the `processedPackageName` in both cases (getting package names from txt file or from output of `ls`).